src/main/java/com/org/jenkins/custom/jenkins/distribution/service/GithubJwtHelper.java

```diff
@@ -0,0 +1,50 @@
+package com.org.jenkins.custom.jenkins.distribution.service;
+
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.Jwts;
+import java.io.File;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Date;
+import com.google.common.io.Files;
+
+
+public class GithubJwtHelper {
+
+    static PrivateKey get(String filename) throws Exception {
+        byte[] keyBytes = Files.toByteArray(new File(filename));
+        PKCS8EncodedKeySpec spec = new PKCS8EncodedKeySpec(keyBytes);
+        KeyFactory kf = KeyFactory.getInstance("RSA");
+        return kf.generatePrivate(spec);
+    }
+
+    static String createJWT(String githubAppId, long ttlMillis) throws Exception {
+        //The JWT signature algorithm we will be using to sign the token
+        SignatureAlgorithm signatureAlgorithm = SignatureAlgorithm.RS256;
+
+        long nowMillis = System.currentTimeMillis();
+        Date now = new Date(nowMillis);
+
+        //We will sign our JWT with our private key
+        Key signingKey = get("jenkins-custom-distribution-bot.der");
```

Comment:
Is this private key the same for all users of the application? Excuse my ignorance, but where does this key come from?

Reply:
This private key comes from the private key of the bot that one creates on github, so when you create an application on github you get a private key for that bot, which you can then sign and verify yourself

Reply:
Is the private key stored in the `.der` file? What is stored in the `.der` file?